Data Structure/Linked List.cpp

```diff
@@ -3,123 +3,140 @@ using namespace std;
 
 struct node
 {
-	int val;
-	node *next;
+    int val;
+    node *next;
 };
 
 node *start;
 
 void insert(int x)
 {
-	node *t=start;
-	if (start!=NULL)
-	{
-			while(t->next!=NULL)
-		{
-			t=t->next;
-		}
-		node *n= new node;
-		t->next=n;
-		n->val=x;
-		n->next=NULL;
-	}
-	else
-	{
-		node *n= new node;
-		n->val=x;
-		n->next=NULL;
-		start=n;
-	}
+    node *t=start;
+    if (start!=NULL)
+    {
+            while(t->next!=NULL)
+        {
+            t=t->next;
+        }
+        node *n= new node;
+        t->next=n;
+        n->val=x;
+        n->next=NULL;
+    }
+    else
+    {
+        node *n= new node;
+        n->val=x;
+        n->next=NULL;
+        start=n;
+    }
 }
 
 void remove(int x){
 
-	if( start == NULL ){
-		cout<<"\nLinked List is empty\n";
-		return ;
-	}
-	else if( start->val == x ){
-		node *temp = start;
-		start = start->next;
-		delete temp;
-		return ;
-	}
-
-	node *temp = start, *parent = start;
-
-	while( temp != NULL && temp->val != x ){
-		parent = temp;
-		temp = temp->next;
-	}
-
-	if( temp == NULL ){
-		cout <<endl <<x <<" not found in list\n";
-		return ;
-	}
-
-	parent->next = temp->next;
-	delete temp;
+    if( start == NULL ){
+        cout<<"\nLinked List is empty\n";
+        return ;
+    }
+    else if( start->val == x ){
+        node *temp = start;
+        start = start->next;
+        delete temp;
+        return ;
+    }
+
+    node *temp = start;
+
+    while( temp != NULL && temp->next->val != x ){
```

Comment:
suppose there is only 1 node in the Linked list, ``temp->next->val`` will cause error
because ``temp->next`` will be ``NULL``